packages/itwinui-react/src/core/Alert/Alert.tsx

```diff
@@ -6,26 +7,135 @@ import cx from 'classnames';
+// Alert.Icon component
+
+type AlertIconOwnProps = {
   /**
-   * Type of the alert.
-   * @default 'informational'
+   * Type of the alert icon.
    */
   type?: 'positive' | 'warning' | 'negative' | 'informational';
+};
+
+const AlertIcon = React.forwardRef((props, ref) => {
+  const { as: Element = 'svg', children, className, type, ...rest } = props;
+
+  if (type) {
+    const StatusIcon = StatusIconMap[type];
+    return (
+      <StatusIcon
+        className={cx('iui-alert-icon', className)}
+        ref={ref}
+        {...rest}
+      />
+    );
+  }
+
+  return (
+    <Element className={cx('iui-alert-icon', className)} ref={ref} {...rest}>
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'svg', AlertIconOwnProps>;
+
+// ----------------------------------------------------------------------------
+// Alert.Message component
+
+type AlertMessageOwnProps = {};
+
+const AlertMessage = React.forwardRef((props, ref) => {
+  const { as: Element = 'span', children, className, ...rest } = props;
+
+  return (
+    <Element className={cx('iui-alert-message', className)} ref={ref} {...rest}>
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'span', AlertMessageOwnProps>;
+
+// ----------------------------------------------------------------------------
+// Alert.ClickableText component
+
+type AlertClickableTextOwnProps = {};
+
+const AlertClickableText = React.forwardRef((props, ref) => {
+  const { as: Element = 'a', children, className, ...rest } = props;
+
+  return (
+    <Element className={cx('iui-alert-link', className)} ref={ref} {...rest}>
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'a', AlertClickableTextOwnProps>;
+
+// ----------------------------------------------------------------------------
+// Alert.CloseButton component
+
+type AlertCloseButtonOwnProps = {
   /**
-   * Text for the link you want to provide.
-   */
-  clickableText?: React.ReactNode;
-  /**
-   * Props for the clickable text. Used for providing `href` and other attributes.
+   * Action handler for close.
    */
-  clickableTextProps?: React.ComponentPropsWithRef<'a'>;
+  onClose: () => void;
+};
+
+const AlertCloseButton = React.forwardRef((props, ref) => {
+  const {
+    as: Element = 'button',
+    children,
+    className,
+    onClose,
+    ...rest
+  } = props;
+
+  return (
+    <Element
+      onClick={onClose}
+      aria-label='Close'
+      type='button'
+      className={cx('iui-alert-button', className)}
+      ref={ref}
+      {...rest}
+    >
+      {children ?? <Alert.CloseButtonIcon />}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'button', AlertCloseButtonOwnProps>;
+
+// ----------------------------------------------------------------------------
+// Alert.CloseButtonIcon component
+
+type AlertCloseButtonIconOwnProps = {};
+
+const AlertCloseButtonIcon = React.forwardRef((props, ref) => {
```

Comment:
it might be possible to remove this and use the generic `<Icon>` component

Reply:
This is making me think we should also use `<Button>` inside `<AlertCloseButton>`. That way, we won't need this line in CSS.

Reply:
also we can use `<Icon>` inside `<Alert.Icon>`

Reply:
Both of these changes have been made resulting in the removal of the `iui-alert-icon` and `iui-alert-button` class names